### DISCUSSION TITLE
Issue ? **runtime error: invalid memory address or nil pointer dereference**

### DISCUSSION THREAD

**Discussion by MarcoEckerlin** [0xJacky/nginx-ui#507] [Q&A] [answered]
Hey Devs,
i dont know how to fix that, i tryed with help of google and chatgpt. Can you help me to solve this?
![image](https://github.com/user-attachments/assets/3a880e70-de31-493a-a744-3eaa5c054b71)

I use **v2.0.0-beta.30** 
on an **Raspberry PI (OS: DietPi)**
and **nginx v1.26.2**
Thanks <3

**Comment by 0xJacky** [comment]
Hi @HeiligesDev,

Could you please provide us with the runtime logs for nginx-ui? 

You can do this by running the following command:

```
journalctl -fu nginx-ui
```

After executing the command, visit the page where you're encountering the error and check the console output.

Thank you!

**Comment by MarcoEckerlin** [comment]
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |     216.185µs |    192.168.2.20 | GET      "/"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |     481.703µs |    192.168.2.20 | GET      "/assets/index-D_sE-xdW.css"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |     334.704µs |    192.168.2.20 | GET      "/assets/useState-C2EiCwdV.js"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |    1.261407ms |    192.168.2.20 | GET      "/assets/collapseMotion-BH7VFSug.js"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |     672.408µs |    192.168.2.20 | GET      "/assets/index-BgmciRcr.js"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |     114.889µs |    192.168.2.20 | GET      "/assets/index-tEwb9DHs.css"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |     176.797µs |    192.168.2.20 | GET      "/assets/SwitchAppearance-D1gnICl9.css"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |      95.796µs |    192.168.2.20 | GET      "/assets/settings-DV_OYkUa.js"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |     969.833µs |    192.168.2.20 | GET      "/assets/dropdown-CnSQABJg.js"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |     365.314µs |    192.168.2.20 | GET      "/assets/index-CC5XlNd7.js"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |     211.129µs |    192.168.2.20 | GET      "/assets/index-Cn1npH7n.js"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |     335.629µs |    192.168.2.20 | GET      "/assets/BaseLayout-B0tNonRR.css"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |     203.574µs |    192.168.2.20 | GET      "/assets/hasIn-DL7I36Dg.js"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |     303.352µs |    192.168.2.20 | GET      "/assets/Terminal-CIVmx5Kd.css"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |    1.891667ms |    192.168.2.20 | GET      "/assets/index-CIpA5JaQ.js"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |     264.592µs |    192.168.2.20 | GET      "/assets/index-B3N_fDbe.js"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |    1.383259ms |    192.168.2.20 | GET      "/assets/dayjs.min-Czz2dMwU.js"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: 2024-08-16 06:28:26        ERROR        middleware/middleware.go:24        runtime error: invalid memory address or nil pointer dereference
> Aug 16 06:28:26 DietPi nginx-ui[1918]: goroutine 155 [running]:
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Nginx-UI/router.InitRouter.Recovery.func2.1()
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/work/nginx-ui/nginx-ui/internal/middleware/middleware.go:23 +0x60
> Aug 16 06:28:26 DietPi nginx-ui[1918]: panic({0x3330d80?, 0x6e81240?})
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /opt/hostedtoolcache/go/1.22.5/x64/src/runtime/panic.go:770 +0x124
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Nginx-UI/internal/user.GetTokenUser({0x4000ff4120, 0x8d})
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/work/nginx-ui/nginx-ui/internal/user/user.go:46 +0x34
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Nginx-UI/router.InitRouter.AuthRequired.func6(0x4001a2af00)
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/work/nginx-ui/nginx-ui/internal/middleware/middleware.go:61 +0x154
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/gin-gonic/gin.(*Context).Next(0x4001a2af00)
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185 +0x40
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Nginx-UI/router.InitRouter.IPWhiteList.func4(0x4001a2af00)
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/work/nginx-ui/nginx-ui/internal/middleware/ip_whitelist.go:14 +0x64
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/gin-gonic/gin.(*Context).Next(0x4001a2af00)
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185 +0x40
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Ng
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 500 |     781.223µs |    192.168.2.20 | GET      "/api/settings/server/name"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: 2024-08-16 06:28:26        ERROR        middleware/middleware.go:24        runtime error: invalid memory address or nil pointer dereference
> Aug 16 06:28:26 DietPi nginx-ui[1918]: goroutine 152 [running]:
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Nginx-UI/router.InitRouter.Recovery.func2.1()
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/work/nginx-ui/nginx-ui/internal/middleware/middleware.go:23 +0x60
> Aug 16 06:28:26 DietPi nginx-ui[1918]: panic({0x3330d80?, 0x6e81240?})
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /opt/hostedtoolcache/go/1.22.5/x64/src/runtime/panic.go:770 +0x124
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Nginx-UI/internal/user.GetTokenUser({0x4000ff43f0, 0x8d})
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/work/nginx-ui/nginx-ui/internal/user/user.go:46 +0x34
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Nginx-UI/router.InitRouter.AuthRequired.func6(0x4001a2af00)
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/work/nginx-ui/nginx-ui/internal/middleware/middleware.go:61 +0x154
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/gin-gonic/gin.(*Context).Next(0x4001a2af00)
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185 +0x40
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Nginx-UI/router.InitRouter.IPWhiteList.func4(0x4001a2af00)
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/work/nginx-ui/nginx-ui/internal/middleware/ip_whitelist.go:14 +0x64
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/gin-gonic/gin.(*Context).Next(0x4001a2af00)
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185 +0x40
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Ng
> Aug 16 06:28:26 DietPi nginx-ui[1918]: 2024-08-16 06:28:26        ERROR        middleware/middleware.go:24        runtime error: invalid memory address or nil pointer dereference
> Aug 16 06:28:26 DietPi nginx-ui[1918]: goroutine 144 [running]:
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Nginx-UI/router.InitRouter.Recovery.func2.1()
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/work/nginx-ui/nginx-ui/internal/middleware/middleware.go:23 +0x60
> Aug 16 06:28:26 DietPi nginx-ui[1918]: panic({0x3330d80?, 0x6e81240?})
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /opt/hostedtoolcache/go/1.22.5/x64/src/runtime/panic.go:770 +0x124
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Nginx-UI/internal/user.GetTokenUser({0x4001958240, 0x8d})
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/work/nginx-ui/nginx-ui/internal/user/user.go:46 +0x34
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Nginx-UI/router.InitRouter.AuthRequired.func6(0x4001825200)
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/work/nginx-ui/nginx-ui/internal/middleware/middleware.go:61 +0x154
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/gin-gonic/gin.(*Context).Next(0x4001825200)
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185 +0x40
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Nginx-UI/router.InitRouter.IPWhiteList.func4(0x4001825200)
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/work/nginx-ui/nginx-ui/internal/middleware/ip_whitelist.go:14 +0x64
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/gin-gonic/gin.(*Context).Next(0x4001825200)
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185 +0x40
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Ng
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 500 |    2.725093ms |    192.168.2.20 | GET      "/api/notifications"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: 2024-08-16 06:28:26        ERROR        middleware/middleware.go:24        runtime error: invalid memory address or nil pointer dereference
> Aug 16 06:28:26 DietPi nginx-ui[1918]: goroutine 167 [running]:
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Nginx-UI/router.InitRouter.Recovery.func2.1()
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/work/nginx-ui/nginx-ui/internal/middleware/middleware.go:23 +0x60
> Aug 16 06:28:26 DietPi nginx-ui[1918]: panic({0x3330d80?, 0x6e81240?})
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /opt/hostedtoolcache/go/1.22.5/x64/src/runtime/panic.go:770 +0x124
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Nginx-UI/internal/user.GetTokenUser({0x40015741b0, 0x8d})
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/work/nginx-ui/nginx-ui/internal/user/user.go:46 +0x34
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Nginx-UI/router.InitRouter.AuthRequired.func6(0x4001421800)
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/work/nginx-ui/nginx-ui/internal/middleware/middleware.go:61 +0x154
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/gin-gonic/gin.(*Context).Next(0x4001421800)
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185 +0x40
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Nginx-UI/router.InitRouter.IPWhiteList.func4(0x4001421800)
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/work/nginx-ui/nginx-ui/internal/middleware/ip_whitelist.go:14 +0x64
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/gin-gonic/gin.(*Context).Next(0x4001421800)
> Aug 16 06:28:26 DietPi nginx-ui[1918]:         /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185 +0x40
> Aug 16 06:28:26 DietPi nginx-ui[1918]: github.com/0xJacky/Ng
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 500 |    3.522481ms |    192.168.2.20 | GET      "/api/otp_secure_session_status"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 500 |    3.098648ms |    192.168.2.20 | GET      "/api/otp_status"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |     148.888µs |    192.168.2.20 | GET      "/assets/index-Dv8uV7LF.js"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |     377.852µs |    192.168.2.20 | GET      "/assets/index-QKRFUHT3.js"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |     544.389µs |    192.168.2.20 | GET      "/assets/environment-B0s-i31O.js"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |      88.166µs |    192.168.2.20 | GET      "/assets/version-BD3EdKen.js"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |      134.24µs |    192.168.2.20 | GET      "/assets/Icon-jPr0U-pG.js"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: [GIN] 2024/08/16 - 06:28:26 | 200 |         637µs |    192.168.2.20 | GET      "/assets/DashBoard-C8PruEXz.css"
> Aug 16 06:28:26 DietPi nginx-ui[1918]: 2024-08-16 06:28:26        ERROR        middleware/middleware.go:24        runtime error: invalid memory address or nil pointer dereference

This is when i open a page from the UI, its there every Page.
Thanks for the fast response.

**Reply by 0xJacky** [reply, marked as answer]
Did try to you visit `http://ip:port/#/install` page in your browser?

**Reply by 0xJacky** [reply]
It appears that nginx-ui is encountering an issue when trying to query data from the database, possibly due to the SQLite file being missing or inaccessible.

**Reply by MarcoEckerlin** [reply]
man thanks <3
Have a nice day
